auth/src/main/java/com/firebase/ui/auth/ui/idp/AuthMethodPickerActivity.java

```diff
@@ -77,7 +83,46 @@ protected void onCreate(Bundle savedInstanceState) {
         mSaveSmartLock = mActivityHelper.getSaveSmartLockInstance();
         findViewById(R.id.email_provider).setOnClickListener(this);
 
-        populateIdpList(mActivityHelper.getFlowParams().providerInfo);
+        Map<String, IdpConfig> providerIdToConfig = new HashMap<>();
+        List<IdpConfig> allConfiguredProviders = mActivityHelper.getFlowParams().providerInfo;
+        for (IdpConfig providerConfig : allConfiguredProviders) {
+            providerIdToConfig.put(providerConfig.getProviderId(), providerConfig);
+        }
+        List<IdpConfig> visibleProviders = new ArrayList<>();
+        if (mActivityHelper.getFlowParams().isReauth) {
+            // display the reauthReason if available
```

Comment:
We should ask UX about this, but instinctively a dialog feels like the "normal" reauth flow.

  1. User clicks a button requiring reauth ("Delete Account" for example)
  2. A dialog shows up that says "This action requires you to sign in again, would you like to continue"
  3. Yes launches the flow, No returns `RESULT_CANCELED`